nix/compat.nix

```diff
@@ -0,0 +1,197 @@
+# Compatibility function to allow flakes to be used by
```

Comment:
Is this taken from somewhere? If so, might be useful to include some info to keep track of it.

Reply:
yes:
https://github.com/edolstra/flake-compat

I can just use a builtin fetch function to grab it instead of dumping it in, I was just in a rush I guess :smile: